### ISSUE TITLE
How to put a new array in a nested object

### ISSUE THREAD

**Issue by bwk1224** [dfahlander/Dexie.js#908]
My question is similar to the issue #560 and I have further questions regarding adding a new array.

Here's what I did.

```js
db.version(1).stores({
    Inspection: 'ID, Date, Data'
});
```

And I load an external json like this.

```js
    const response = await fetch('../template.json');
    const json = await response.json();

    db.on("populate", function () {
        db.Inspection.add({ ID: "Template_1", Date: Date.now(), Data: json });
    });
```

A sample of my json structure is like this.

```json
{
   "Component":[
      {
         "GUID":"11111",
         "ComponentTitle":"2nd Floor",
         "Section":[
            {
               "GUID":"abc",
               "Title":"Area 1",
               "Status":""
            },
            {
               "GUID":"def",
               "Title":"Area 2",
               "Status":""
            }
         ]
      },
      {
         "GUID":"22222",
         "ComponentTitle":"3rd Floor",
         "Section":[
            {
               "GUID":"ghr",
               "Title":"Area 1",
               "Status":""
            },
            {
               "GUID":"jkl",
               "Title":"Area 2",
               "Status":""
            }
         ]
      }
   ]
}
```

So far I was able to change values of nested objects after referencing #560 like this.

db.Inspection.update("Template_1", { "Data.Component.0.Section.1.Status": "Checked" });

But I get an error when I try to put new array like this.
db.Inspection.put("Template_1", { "Data.Component.0.Section": {"GUID":"ddddd", "Title":"Area 3", "Status":"" }});

The error msg is this.
Unhandled rejection: DataError: Failed to execute 'put' on 'IDBObjectStore': The object store uses in-line keys and the key parameter was provided.
Error: Failed to execute 'put' on 'IDBObjectStore': The object store uses in-line keys and the key parameter was provided.

I am new to indexedDB and Dexie. Any help would be appreciated. Thanks!

**Comment by dfahlander** [comment]
[Table.put()](https://dexie.org/docs/Table/Table.put()) takes a single argument in your case (as you have inline keys (which is same as [inbound keys](https://dexie.org/docs/inbound)). Maybe you want to use [Table.update()](https://dexie.org/docs/Table/Table.update()) instead?

**Comment by bwk1224** [comment]
Thanks for a quick reply. It's good to know Table.put() replaces the entire 'row'. I was under the impression that put() will either update or create a new row as needed.

My code was trying to add a new array to an object without array format. So I had to update my json format first, and then I was able to fix the issue by referring to your suggestion from the other thread.
`db.inspections.where('id').equals(id).modify(x => x.tags.skill[0].third.push({special3:'run'}) );`

I'll close the thread.
